app/javascript/packs/articlePage.jsx

```diff
@@ -20,48 +23,99 @@ if (snackZone) {
 // eslint-disable-next-line no-restricted-globals
 top.addSnackbarItem = addSnackbarItem;
 
+function hideCopyLinkAnnouncerIfVisible() {
+  document.getElementById('article-copy-link-announcer').hidden = true;
+}
+
+// Initialize the share options
+const shareDropdownButton = document.getElementById('article-show-more-button');
+
+if (shareDropdownButton.dataset.initialized !== 'true') {
+  if (Runtime.isNativeAndroid('shareText')) {
+    // Android native apps have enhanced sharing capabilities for Articles and don't use our standard dropdown
+    shareDropdownButton.addEventListener('click', () =>
+      AndroidBridge.shareText(location.href),
+    );
```

Comment:
@fdoxyz - just a heads up that this code has moved from initializers to packs. The code itself hasn't changed though, so I don't believe there's a great regression risk. Do you think we need to specifically test this PR in terms of the android mobile app where this behavior is triggered?

Reply:
Thank you for the heads up @aitchiss! I tested this locally with the Android DEV app and things are working as expected.

> Do you think we need to specifically test this PR in terms of the android mobile app where this behavior is triggered?

I think it would be a great addition to avoid regressions! Would it be difficult to add a test case where we stub `Runtime.isNativeAndroid('shareText')` to return `true` and then check that `navigator.share` is called when a click to the `shareDropdownButton` is clicked/tapped?

I'm asking because it's non-trivial for me to work my way round stubbing these in Cypress tests (especially the `navigator`) 😅  If you do manage to add this test case I'd appreciate if you ping me on it, just to take a look and make sure I learn how these stubs can be done in Cypress for future spec I'll nee to write 🙏 

The DEV Android app isn't a major focus right now, but future Android native efforts will rely on this feature as well, at least until `navigator.share` is supported within the Android WebView

Reply:
I was full of hope about stubbing the various things in Cypress, but it looks like it's not so straightforward 🙃 

I think probably it could be possible with more investigation, but maybe that would need to be separate to this PR as it might take a while! I'm thinking that eventually what would be great is if the android app had its own test for this.

Reply:
> A small/medium problem of this approach would be that the Android app checks wouldn't be run in forem/forem PRs which could potentially break this behavior.

Yeah that's very true. Agree that it's definitely worth some research - let me know if I can help at all!